### ISSUE TITLE
User file showing fatal error when sizing hot water coil UA

### ISSUE THREAD

**Issue by rraustad** [NREL/EnergyPlus#6088]
Issue overview
--------------
Very low zone loads cause the hot water coil air flow rate used to size the coil UA to be lower than SmallAirVolFlow (0.001) which results in the sizing routine setting the air flow rate to 0.

    if ( IsAutoSize || SizingDesRunThisAirSys || SizingDesRunThisZone ) {
        if ( AutosizeDes < SmallAirVolFlow && bCheckForZero) {
            AutosizeDes = 0.0;
        }
    }

Suggest adding the Boolean that avoids the low air flow check (2 places):

    } else if ( SizingType == HeatingAirflowUASizing ) {
        if ( TermUnitSingDuct || TermUnitPIU || TermUnitIU ) {
            AutosizeDes = StdRhoAir * TermUnitSizing( CurZoneEqNum ).AirVolFlow * TermUnitSizing(CurZoneEqNum ).ReheatAirFlowMult;
        } else if ( ZoneEqFanCoil ) {
            AutosizeDes = StdRhoAir * FinalZoneSizing( CurZoneEqNum ).DesHeatVolFlow;
        } else {
            AutosizeDes = FinalZoneSizing( CurZoneEqNum ).DesHeatMassFlow;
        }
        bCheckForZero = false;

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (E+ Dev support defect file folder 6088)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

Tampa weather (Miami shows issue)

**Comment by rraustad** [comment]
Made suggested correction above and user file runs to completion. However, coils sizes are very small (30W) and air flow used during sizing is also very small (0.0009 m3/s or 2 CFM). Coil UA sizes out to 1W/K, seems very unreasonable.

    ! <Zone Sizing Information>, Zone Name, Load Type, Calc Des Load {W}, User Des Load {W}, Calc Des Air Flow Rate {m3/s}, User Des Air Flow Rate {m3/s}, Design Day Name, Date/Time of Peak, Temperature at Peak {C}, Humidity Ratio at Peak {kgWater/kgDryAir}, Floor Area {m2}, # Occupants, Calc Outdoor Air Flow Rate {m3/s}, Calc DOAS Heat Addition Rate {W}
    Zone Sizing Information, TZ15-34 ZONE, Heating, 15.69116, 19.61396, 7.10777E-004, 7.73994E-003, TAMPA INTERNATIONAL AP ANN HTG 99.6% CONDNS DB, 1/21 24:00:00, 3.60000, 4.89651E-003, 30.47220, 0.32800, 7.73994E-003, 0.00000
    Zone Sizing Information, TZ15-41 ZONE, Heating, 13.59024, 16.98780, 6.15609E-004, 0.32848, TAMPA INTERNATIONAL AP ANN HTG 99.6% CONDNS DB, 1/21 24:00:00, 3.60000, 4.89651E-003, 64.66052, 34.80000, 0.32848, 0.00000
    Zone Sizing Information, TZ15-51 ZONE, Heating, 2.53462, 3.16828, 1.14813E-004, 4.01391E-003, TAMPA INTERNATIONAL AP ANN HTG 99.6% CONDNS DB, 1/21 24:00:00, 3.60000, 4.89651E-003, 7.52515, 0.42525, 4.01391E-003, 0.00000


     Component Sizing Information, Coil:Heating:Water, TZ15-34 ZONE CV HW RHT COIL, Design Size Design Coil Load [W], 30.77731
     Component Sizing Information, Coil:Heating:Water, TZ15-34 ZONE CV HW RHT COIL, Design Size Maximum Water Flow Rate [m3/s], 3.67747E-007
     Component Sizing Information, Coil:Heating:Water, TZ15-34 ZONE CV HW RHT COIL, Design Size U-Factor Times Area Value [W/K], 1.16591
     
     Component Sizing Information, Coil:Heating:Water, TZ15-41 ZONE CV HW RHT COIL, Design Size Design Coil Load [W], 26.63090
     Component Sizing Information, Coil:Heating:Water, TZ15-41 ZONE CV HW RHT COIL, Design Size Maximum Water Flow Rate [m3/s], 3.18203E-007
     Component Sizing Information, Coil:Heating:Water, TZ15-41 ZONE CV HW RHT COIL, Design Size U-Factor Times Area Value [W/K], 1.00829
     
     Component Sizing Information, Coil:Heating:Water, TZ15-51 ZONE CV HW RHT COIL, Design Size Design Coil Load [W], 4.96814
     Component Sizing Information, Coil:Heating:Water, TZ15-51 ZONE CV HW RHT COIL, Design Size Maximum Water Flow Rate [m3/s], 5.93626E-008
     Component Sizing Information, Coil:Heating:Water, TZ15-51 ZONE CV HW RHT COIL, Design Size U-Factor Times Area Value [W/K], 0.18813


I guess this is OK for a simulation but design firms would never install such a small coil. Maybe there should be minimum sizes used for terminal unit heating coils? Other coil types? New object to show minimum capacity or air flow of coils during sizing? The easy fix is not always the correct fix.

**Comment by rraustad** [comment]
See PR #6089.

**Comment by mjwitte** [comment]
The original error that the user saw is rather frustrating:
```
** Severe ** Autosizing of heating coil UA failed for Coil:Heating:Water "TZ15-41 ZONE CV HW RHT COIL"
** ~~~ ** Bad starting values for UA
** ~~~ ** Lower UA estimate = 2.663089E-002 W/m2-K (0.1% of Design Coil Load)
** ~~~ ** Upper UA estimate = 26.630895 W/m2-K (100% of Design Coil Load)
** ~~~ ** Zone "TZ15-41 ZONE" coil sizing conditions (may be different than Sizing inputs):
** ~~~ ** Coil inlet air temperature = 12.220 C
** ~~~ ** Coil inlet air humidity ratio = 8.000E-003 kgWater/kgDryAir
** ~~~ ** Coil inlet air mass flow rate = 0.395304 kg/s
** ~~~ ** Design Coil Capacity = 0.000 W
** ~~~ ** Design Coil Load = 26.630 W
** ~~~ ** Inadequate water side capacity: in Plant Sizing for this hot water loop
** ~~~ ** increase design loop exit temperature and/or decrease design loop delta T
** ~~~ ** Plant Sizing object = HW LOOP
** ~~~ ** Plant design loop exit temperature = 65.560 C
** ~~~ ** Plant design loop delta T = 20.000 C
** Fatal ** Preceding water coil sizing errors cause program termination
```
@rraustad Is there any way this can be changed to do something reasonable instead of a fatal?

**Comment by rraustad** [comment]
Yes, we need to be smarter on this one.  The problem is the coil capacity is 0 while the load is > 0. Whether that's due to air flow sizing or delta conditions capacity sizing doesn't matter, it's an issue with the choice to size the UA. 

** ~~~ ** Design Coil Capacity = 0.000 W
** ~~~ ** Design Coil Load = 26.630 W

**Comment by rraustad** [comment]
This is probably the issue where water coil air flow is set to 0 if < 0.001 when attempting to calc UA. My fix corrects but we still need to discuss.

**Comment by rraustad** [comment]
Regarding the frustrating message, I could check to see if air and water flow rate > 0 and if not set UA = 1 ? In this run I also reverted back to checking for the 2 cfm limit (really, that's our limit). I should tell the user something if RegulaFalsi fails, no? The only thing missing in that message is air flow rate. That message should report much less now (i.e., RegulaFalsi shouldn't fail with -2).

The results show the first 2 coils were sized, and the last bypassed the UA calc and just set UA = 1. The sizes for 2nd and 3rd coil don't seem any different. So why aren't 2 and 3 different? I guess it's because we don't report the air flow rate, and for 3 it was 0. So coil 2 must have an air flow rate > 0.001 m3/s (2.12 CFM). Yup, 0.00107 m3/s. It's a good thing we sized that one and not the other!

    Component Sizing Information, Coil:Heating:Water, TZ15-33 ZONE CV HW RHT COIL, Design Size Design Coil Load [W], 231.95380
    Component Sizing Information, Coil:Heating:Water, TZ15-33 ZONE CV HW RHT COIL, Design Size Maximum Water Flow Rate [m3/s], 2.77153E-006
    Component Sizing Information, Coil:Heating:Water, TZ15-33 ZONE CV HW RHT COIL, Design Size U-Factor Times Area Value [W/K], 124.16552
    Component Sizing Information, Coil:Heating:Water, TZ15-34 ZONE CV HW RHT COIL, Design Size Design Coil Load [W], 30.77731
    Component Sizing Information, Coil:Heating:Water, TZ15-34 ZONE CV HW RHT COIL, Design Size Maximum Water Flow Rate [m3/s], 3.67747E-007
    Component Sizing Information, Coil:Heating:Water, TZ15-34 ZONE CV HW RHT COIL, Design Size U-Factor Times Area Value [W/K], 1.16591
    Component Sizing Information, Coil:Heating:Water, TZ15-41 ZONE CV HW RHT COIL, Design Size Design Coil Load [W], 26.63090
    Component Sizing Information, Coil:Heating:Water, TZ15-41 ZONE CV HW RHT COIL, Design Size Maximum Water Flow Rate [m3/s], 3.18203E-007
    Component Sizing Information, Coil:Heating:Water, TZ15-41 ZONE CV HW RHT COIL, Design Size U-Factor Times Area Value [W/K], 1.00000

**Comment by Myoldmopar** [comment]
Closed via #6089
